### PR TITLE
See if refusal attribute exists in ChatCompletionMessage before referencing it

### PR DESCRIPTION
**Pull request by callmephilip**
Instructor is currently broken when using Litellm (i suspect they are behind on supporting `refusal` attr or smth). I am trying to fix this with a quick check before referencing `refusal` attribute

----

| :rocket: | This description was created by [Ellipsis](https://www.ellipsis.dev) for commit cde3b7011a77f98a20554898c88eff4b29aea201  | 
|--------|--------|

### Summary:
Added a check for the `refusal` attribute in `parse_tools` function to prevent errors when it's missing in `ChatCompletionMessage`.

**Key points**:
- Added `hasattr` check for `refusal` attribute in `parse_tools` function in `instructor/function_calls.py`.
- Prevents errors when `refusal` attribute is missing in `ChatCompletionMessage`.
- Specifically addresses compatibility with tools like `litellm`.

----
Generated with :heart: by [ellipsis.dev](https://www.ellipsis.dev)

### CODE REVIEW

**Review by ivanleomk — CHANGES_REQUESTED** [review]

**Comment by brycedrennan** [comment]
Thanks for addressing this @callmephilip! Just found Instructor, awesome library, but immediately ran into this bug.

File: instructor/function_calls.py [hunk 1]
```diff
@@ -315,9 +315,12 @@ def parse_tools(
         assert (
             len(message.tool_calls or []) == 1
         ), "Instructor does not support multiple tool calls, use List[Model] instead."
-        assert (
-            message.refusal is None
-        ), f"Unable to generate a response due to {message.refusal}"
+        # this field seems to be missing when using instructor with some other tools (e.g. litellm)
+        # trying to fix this by adding a check
+        if hasattr(message, "refusal"):
```
[the remark this answers is not in the document]
Comment (callmephilip) [reply instructor/function_calls.py:314]: done

File: instructor/function_calls.py [hunk 2]
```diff
@@ -315,9 +315,12 @@ def parse_tools(
...
+            assert (
+                message.refusal is None
+            ), f"Unable to generate a response due to {message.refusal}"
```
[the remark this answers is not in the document]
Comment (callmephilip) [reply instructor/function_calls.py:317]: separated this into 2 separate tests + reused `test_model` via fixtures

[the remark this answers is not in the document]
Comment (ivanleomk) [reply instructor/function_calls.py:317]: Looks good!

**Review by ivanleomk — APPROVED** [review]
@callmephilip This PR looks good to me. Thanks for submitting the change, going to merge things in now! :)

**Pull request merged**